default_policy.tf

```
@@ -0,0 +1,290 @@
+data "aws_iam_policy_document" "default_policy" {
```

```suggestion
data "aws_iam_policy_document" "this" { # or `controller`
```
We tend to name TF resources that are used only once using `this`. If there are multiple instances then name it accordingly. In this case `this` should work.

---

Also, we should add a condition using `count = var.enabled && var.irsa_policy == null` to prevent unnecessary creation of this when it wont be used after all.

---

Using `this` with condition as suggested.

---

fixed.